// Copyright 2015 gRPC authors.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

package gst_task;

service Greeter {
  rpc StartTask (StartRequest) returns (StartReply) {}
  rpc StopTask (StopRequest) returns (StopReply) {}
}

message StartRequest {
  string file_name = 1;
}

message StartReply {
  int32 status_code = 1;
  string service_id = 2;
}

message TaskReply{
    int32 status_code = 1;
    string msg = 2;
}

message TaskRequest{
    string service_id = 1;
}

message SetTaskPropRequest{
    string service_id = 1;
    repeated SetPropRequest =2;
}

message SetPropRequest{
    string element_name =1;
    string prop_name =2;
    string value = 3;
}

message RunRequest {
  string service_id = 1; 
}

message RunReply {
  int32 status_code = 1;
  bytes chunk = 1; 
}